rls/src/test/java/io/grpc/rls/LinkedHashLruCacheTest.java

```diff
@@ -107,9 +103,11 @@ public void eviction_expire() {
     cache.cache(1, survivor);
 
     fakeClock.forwardTime(10, TimeUnit.NANOSECONDS);
+    cache.cleanupExpiredEntries();
```

Comment:
Seems like you should be using cache.fitToLimit() at least some of the time (or maybe it deserves its own test case).

Reply:
That isn't related to this PR though, right? The only part of this PR is `cleanupExpiredEntries()` moved to the consumer of the class, so the timer was no longer set automatically.

Reply:
Yes, everything specifically related to the PR is fine, it's just that there is a lot of cleanup that should be done nearby in the code.

Reply:
"nearby" code is pretty much the whole file, because I un-indented text by removing the lock.

I've created https://github.com/grpc/grpc-java/issues/11243 for us to drop wish-list items, because as we touch things we'll find more. (That's how this was created.)